src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -928,13 +964,17 @@ private void EnterParameter(ParameterSymbol parameter, TypeSymbolWithAnnotations
             {
                 if (EmptyStructTypeCache.IsTrackableStructType(parameterType.TypeSymbol))
                 {
-                    InheritNullableStateOfTrackableStruct(parameterType.TypeSymbol, slot, valueSlot: -1, isByRefTarget: parameter.RefKind != RefKind.None, slotWatermark: GetSlotWatermark());
+                    InheritNullableStateOfTrackableStruct(
+                        parameterType.TypeSymbol,
+                        slot,
+                        valueSlot: -1,
+                        isDefaultValue: parameter.ExplicitDefaultConstantValue?.IsNull == true,
```

Comment:
>isDefaultValue: parameter.ExplicitDefaultConstantValue?.IsNull == true, [](start = 24, length = 71)

Is this going to properly handle structures?

Reply:
`parameter.ExplicitDefaultConstantValue.IsNull` is `true` for `default(S)` and `new S()`.

---
In reply to: [246951009](https://github.com/dotnet/roslyn/pull/32304#discussion_r246951009) [](ancestors = 246951009)

Reply:
Added a missing default value test, and checked the implementations of `ExplicitDefaultConstantValue` look reasonable.

---
In reply to: [246959155](https://github.com/dotnet/roslyn/pull/32304#discussion_r246959155) [](ancestors = 246959155,246955015,246951009)